src/main/java/com/minecolonies/coremod/util/LanguageHandler.java

```diff
@@ -33,17 +35,84 @@ private LanguageHandler()
      */
     public static void sendPlayerLocalizedMessage(@NotNull final EntityPlayer player, final String key, final Object... args)
     {
-        sendPlayerMessage(player, format(key, args));
+        sendPlayerMessage(player, key, args);
     }
 
     /**
      * Send a message to the player.
      * @param player the player to send to.
+     * @param key the key of the message.
      * @param message the message to send.
      */
-    public static void sendPlayerMessage(@NotNull final EntityPlayer player, final String message)
+    public static void sendPlayerMessage(@NotNull final EntityPlayer player, final String key, final Object... message)
     {
-        player.addChatComponentMessage(new TextComponentString(message));
+        TextComponentTranslation translation = null;
+
+        int onlyArgsUntil = 0;
+        for (Object object : message)
+        {
+            if (object instanceof ITextComponent || object instanceof TextComponentTranslation)
+            {
+                if(onlyArgsUntil == 0)
+                {
+                    onlyArgsUntil = -1;
+                }
+                break;
+            }
+            onlyArgsUntil++;
+        }
+
+        if (onlyArgsUntil >= 0)
+        {
+            Object[] args = new Object[onlyArgsUntil];
+            for(int i = 0; i < onlyArgsUntil; i++)
+            {
+                args[i] = message[i];
+            }
+            translation = new TextComponentTranslation(key, args);
+        }
+
+        for (Object object : message)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'object' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)